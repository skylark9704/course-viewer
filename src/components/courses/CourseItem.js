import React from "react";
import { Link } from "react-router-dom";
function CourseItem(props) {
  const { title, category, authorName, slug } = props.course;
  const route = `/course/${slug}`;
  return (
    <tr>
      <td>
        <button className="btn btn-light btn-small">Watch</button>
      </td>
      <td>
        <Link to={route}>{title}</Link>
      </td>
      <td>{authorName}</td>
      <td>{category}</td>
      <td><button className="btn btn-outline-danger">Delete</button></td>
    </tr>
  );
}

export default CourseItem
