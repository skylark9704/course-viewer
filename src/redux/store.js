import { createStore, applyMiddleware, compose } from "redux";
import courseReducer from "./reducers";
import ReduxThunk from "redux-thunk";
import { devToolsEnhancer } from "redux-devtools-extension";

let store = createStore(
  courseReducer,
  compose(applyMiddleware(ReduxThunk), devToolsEnhancer())
);

export default store;
