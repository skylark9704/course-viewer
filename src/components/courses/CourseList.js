import React from "react";
import CourseItem from "./CourseItem";
function CourseList(props) {
  const { items } = props;
  return (
    <table className="table
    ">
      <thead>
        <tr>
          <th></th>
          <th>
            <b>Title</b>
          </th>
          <th>
            <b>Author</b>
          </th>
          <th>
            <b>Category</b>
          </th>
        </tr>
      </thead>

      <tbody>
        {items &&
          items.map((elem, index) => (
            <CourseItem
              course = {elem}
              key={index}
            />
          ))}
      </tbody>
    </table>
  );
}
export default CourseList;
