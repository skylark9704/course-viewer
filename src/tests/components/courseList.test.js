import React from "react";
import CourseList from "../../components/courses/CourseList";
import CourseItem from "../../components/courses/CourseItem";
import { Provider } from "react-redux";
import store from "../../redux/store";
import { within, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

const component = (
  <Provider store={store}>
    <Router>
      <CourseList
        items={[
          {
            author: "author",
            authorId: 2,
            title: "title",
            category: "category",
            slug: "slug"
          }
        ]}
      />
    </Router>
  </Provider>
);

const tree = render(component);

describe("Test CourseList Component", () => {
  it("should render", () => {
    expect(tree.firstChild).toMatchSnapshot();
  });

  it('should contain table', () => {
      const { getAllByTestId } = render(component)
      const table = getAllByTestId("courses-table")
      expect(table.length).toBe(1)
  });

  it('should contain a table head', () => {
    const { getAllByTestId } = render(component)
    const tableHead = getAllByTestId("courses-table-head")
    expect(tableHead.length).toBe(1)
  });

  it('should contain a table body', () => {
    const { getAllByTestId } = render(component)
    const tableBody = getAllByTestId("courses-table-body")
    expect(tableBody.length).toBe(1)
  });
});
