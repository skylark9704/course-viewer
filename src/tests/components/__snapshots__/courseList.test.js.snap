// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CourseList Component should render 1`] = `undefined`;
